src/Razor/src/Microsoft.CodeAnalysis.Remote.Razor/Microsoft.CodeAnalysis.Remote.Razor.csproj

```diff
@@ -24,9 +24,15 @@
 
   <ItemGroup>
     <PackageReference Include="Microsoft.AspNetCore.Razor.Language" Version="$(MicrosoftAspNetCoreRazorLanguagePackageVersion)" />
-    <PackageReference Include="Microsoft.CodeAnalysis.Remote.ServiceHub" Version="$(Tooling_MicrosoftCodeAnalysisRemoteServiceHubPackageVersion)" PrivateAssets="All" />
-    <PackageReference Include="Microsoft.CodeAnalysis.ExternalAccess.Razor" Version="$(Tooling_MicrosoftCodeAnalysisExternalAccessRazorPackageVersion)" />
     <PackageReference Include="Nerdbank.Streams" Version="$(NerdbankStreamsPackageVersion)" />
+
+    <!--
+      Microsoft.CodeAnalysis.Remote.ServiceHub brings in a transitive dependency to StreamJsonRpc of 2.7.X. We found that in practice this would flakily break our
+      CI builds here: https://github.com/dotnet/razor-tooling/issues/4327
+      Now we aren't sure why this exposes a "flaky" issue; however, to workaround the break we pin StreamJsonRpc.
```

Comment:
Nice! Also wondering why this would be flaky, but happy to have a resolution regardless.

We've had a couple of other packages (unfortunately forget which ones) which seem to run into similar problems, should we pin those too?

Reply:
> We've had a couple of other packages (unfortunately forget which ones) which seem to run into similar problems, should we pin those too?

Assuming pinning works I'd say "yes"; however, if we find ourselves hitting this a lot we may end up finding a lot of value into digging into the issue to understand "why" things are breaking